### PR TITLE
feat: Support event headers as GET param [DHIS2-12314]

### PR DESCRIPTION
**Pull request by maikelarabori**
This feature is about supporting a new `headers` param in _events endpoints_.
This new param will allow the clients to choose the `headers` they want back as part of the response.
The `headers` ordering in the response will match the same order of the `headers` specified as the URL parameter.
This should work for `/events/query` and `enrollments/query` endpoints.

Examples of **GET** requests using the `headers` parameter:

`http://localhost:8080/dhis/api/29/analytics/events/query/eBAyeGv0exc.json?dimension=pe:LAST_12_MONTHS&dimension=ou:jUb8gELQApl;TEQlaapDQoK;eIQbndfxQMb;Vth0fbpFcsO;PMa2VCrupOd;O6uvpzGd5pu;bL4ooGhyHRQ;kJq2mPyFEHo;fdc6uOvgoji;at6UHUQatSo;lc3eMKXaEfw;qhqAxPSTUXp;jmIPBj66vD6&stage=Zj7UnCAulEk&displayProperty=NAME&outputType=EVENT&desc=eventdate&pageSize=100&page=1&headers=ouname,oucode,storedby`

`http://localhost:8080/dhis/api/29/analytics/enrollments/query/WSGAb5XwJ3Y.json?dimension=pe:LAST_12_MONTHS&dimension=ou:ImspTQPwCqd&dimension=edqlbukwRfQ[0].vANAXwtLwcT&dimension=edqlbukwRfQ[-1].vANAXwtLwcT&dimension=edqlbukwRfQ[2].vANAXwtLwcT&stage=edqlbukwRfQ&displayProperty=NAME&outputType=ENROLLMENT&desc=enrollmentdate&pageSize=100&page=1&skipMeta=true&headers=storedby,ouname,edqlbukwRfQ[-1].vANAXwtLwcT,edqlbukwRfQ[2].vANAXwtLwcT,pi,tei,enrollmentdate,incidentdate,vANAXwtLwcT`

### CODE REVIEW

File: dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/grid/ListGrid.java [hunk 1]
```diff
@@ -1092,6 +1098,92 @@ public Grid addRows( SqlRowSet rs )
     // Supportive methods
     // -------------------------------------------------------------------------
 
+    /**
+     * This method will take a Grid and keep only the given list of headers. All
+     * other GridHeaders and respective columns will be removed from the Grid.
+     *
+     * @param headers
+     */
+    public void keepOnlyThese( final Set<String> headers )
```
Comment (d-bernat) [inline dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/grid/ListGrid.java:1107]: please use Override annotation

File: dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/grid/ListGrid.java [hunk 2]
```diff
@@ -1092,6 +1098,92 @@ public Grid addRows( SqlRowSet rs )
...
+
+    /**
+     * Re-order the GridHeaders of the given Grid based on the List headers. The
+     * final Grid will have the all its headers defined in the same order as the
+     * given List of headers.
+     *
+     * @param headers
+     * @return a Set of indexes that holds the holds the new order
+     */
+    public Set<Integer> repositionHeaders( final Set<String> headers )
```
Comment (d-bernat) [inline dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/grid/ListGrid.java:1132]: please use Override annotation

File: dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/grid/ListGrid.java [hunk 3]
```diff
@@ -1092,6 +1098,92 @@ public Grid addRows( SqlRowSet rs )
...
+    }
+
+    /**
+     * Based on the given column indexes, this method will order the current
+     * columns in the Grid. The new positions of the columns will respect the
+     * new indexes.
+     *
+     * @param newColumnsIndexes
+     */
+    public void repositionColumns( final Set<Integer> newColumnsIndexes )
```
Comment (d-bernat) [inline dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/grid/ListGrid.java:1167]: please use Override annotation

**Review by d-bernat — APPROVED** [review]
Great job, just minor issues to add (@Override)

File: dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/grid/ListGrid.java (hunk 1 above)
Comment (maikelarabori) [reply dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/grid/ListGrid.java:1108]: Done

File: dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/grid/ListGrid.java (hunk 3 above)
Comment (maikelarabori) [reply dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/grid/ListGrid.java:1170]: Done

File: dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/grid/ListGrid.java (hunk 2 above)
Comment (maikelarabori) [reply dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/grid/ListGrid.java:1134]: Done

**Review by gnespolino — APPROVED** [review]

**Pull request merged**